### DISCUSSION TITLE
Configuring transition behavior during raster plan

### DISCUSSION THREAD

**Discussion by dpiet** [tesseract-robotics/tesseract_planning#251] [Q&A]
When transitioning between raster segments, what is the recommended way for configuring the planner behavior such that the TCP lifts off the workpiece some distance before freespace planning to the next segment?

**Comment by Levi-Armstrong** [comment]
Usually I added a retract and approach point to each raster.

**Comment by Simon248** [comment]
Hello, 

What if I want to have distinct trajectories for approach / process / retreat.
For instance if i've got a process I want to start on the raster and stop on approach / retreat (painting, welding, NDC...)

I imagine adding approach and retreat points in the raster will result in one single trajectory and I will loose distinction between those 3 phases.

I've got to do that and i see 2 others options:
     - Create a new [raster_motion_task](https://github.com/tesseract-robotics/tesseract_planning/blob/master/tesseract_task_composer/planning/src/nodes/raster_motion_task.cpp)  which handle approach / retreat. I'm not sure about the way to go, what to modify.
     - Using a kind of instruction (i did not dig into how to use instruction at the moment)

Any other strategy? Which one seems the best for a quick implementation and later for a more robust one?

Thanks.

**Reply by Levi-Armstrong** [reply]
My recommendation is to implement a new raster_motion_task which supports this. The way it was done in the past the composite for the process would a composite containing three composites (approach, process, retreat). I would be supportive of merging this into the repository if you wanted to take this on.

**Comment by Simon248** [comment]
ok, i will give a try as soon as i can, it seems to be a good way to get a better understanding of the pipe mechanism.
thanks.
